rskj-core/src/main/java/co/rsk/peg/Federation.java

```diff
@@ -54,7 +55,7 @@ public Federation(List<FederationMember> members, Instant creationTime, long cre
         // effectively immutable
         this.members = Collections.unmodifiableList(members.stream().sorted(FederationMember.BTC_RSK_MST_PUBKEYS_COMPARATOR).collect(Collectors.toList()));
 
-        this.creationTime = creationTime;
+        this.creationTime = creationTime.truncatedTo(ChronoUnit.MILLIS);
```

Comment:
I'm a bit confused by this change. Why can't we directly assign the `creationTime` as parameterised?

Reply:
I did leave a comment here about this change, but seems it's disappeared after I've update the PR.

The issue here is that in java11 nanoseconds part of a `Instant` object is not zero. In the `BridgeSerializationUtils` class we have two helper methods `serializeFederationWithSerializer` and `deserializeFederationWithDesserializer`. And they do not take into account nanoseconds while serializing / deserializing `Federation` objects (e.g the `serializeFederationWithSerializer` method uses this approach `federation.getCreationTime().toEpochMilli()`, so only milliseconds are being taken and nanos are ignored). Without this change unit tests fail, as serialized `Federation` object after deserializing does not equal to original one.

Also not sure if we can serialize nanos, as older nodes and the ones run by jvn8 would not understand it

Reply:
thanks for the clarifications